tajo-client/src/main/java/org/apache/tajo/client/SessionConnection.java

```diff
@@ -87,6 +87,10 @@ public SessionConnection(ServiceTracker tracker, @Nullable String baseDatabase,
     this.properties = properties;
 
     this.manager = RpcClientManager.getInstance();
+    this.manager.setRetries(properties.getInt(RpcConstants.RPC_CLIENT_RETRY_MAX, RpcConstants.DEFAULT_RPC_RETRIES));
+    this.manager.setTimeoutSeconds(
+        properties.getInt(RpcConstants.RPC_CLIENT_TIMEOUT_SECS, 0)); // disable rpc timeout
```

Comment:
IMO, It should be handled in session timeout. 
If timeout is set to properties, it will work